### PR TITLE
Document Go job worker usage

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR adds documentation for the Go job worker, including for the new streaming job worker. It seems there was none at all prior to this, so I included some generic documentation based on the Java client's.

## When should this change go live?

- [x] This change is not yet live and should not be released until 8.4.0 release
- [ ] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [x] This functionality is already (partially) available but undocumented
- [ ] This is a bug fix or security concern.

## PR Checklist

- [ ] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [ ] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/components/concepts/job-workers.md
Comment (npepinpe) [reply docs/components/concepts/job-workers.md:113]: Maybe I need a new line so it stops trying to fix it like this.

**Review by Zelldon — APPROVED** [review]
Thanks @npepinpe for writing this 👍🏼 Looks good to me, just smaller minor things.

File: docs/apis-tools/go-client/job-worker.md
Comment (npepinpe) [reply docs/apis-tools/go-client/job-worker.md:19]: Yes. Do you think users should care about that, or it's more implementation detail?

File: docs/apis-tools/go-client/job-worker.md
Comment (npepinpe) [reply docs/apis-tools/go-client/job-worker.md:23]: It's the default value, not a constant. I'll replace it. This was copied from Java where the threshold is hard coded I think.

File: docs/apis-tools/go-client/job-worker.md
Comment (npepinpe) [reply docs/apis-tools/go-client/job-worker.md:8]: Probably to some extent. I'll let our technical writing team decide on that :upside_down_face:

File: docs/apis-tools/go-client/job-worker.md
Comment (npepinpe) [reply docs/apis-tools/go-client/job-worker.md:111]: Yeah. But the max message size does give you some upper bound, whereas the payload is variable sized, so if you can use max message size for the worst case scenario.

Comment (npepinpe) [reply docs/apis-tools/go-client/job-worker.md]: I rephrased it, hopefully it's better.

**Review by Zelldon — DISMISSED** [review]
🚀

**Comment by akeller** [comment]
@christinaausley looks like a review from you is the last step

**Review by christinaausley — APPROVED** [review]
Nice work on this 👍

**Comment by pepopowitz** [comment]
Two things I noticed with this PR: 

1. The merge timing between this & was such that one of the links is broken (from the new `go-client/job-worker` page, to `[gRPC technical error handling](/apis-tools/grpc.md#technical-error-handling)`). This is because #3150 reorganized the target of that link into a standalone page, and this branch didn't update before merging. None of this is a result of anyone doing something wrong, just bad timing! I'll fix this in a PR immediately, because it's preventing the docs from building.
2. The new `go-client/job-worker` page is not added to `sidebars.js`, so it's not appearing in the navigation, and when you navigate to it directly it shows no sidebar navigation. I suspect this is an oversight. I'll leave it up to @npepinpe or @christinaausley to resolve this!

**Comment by npepinpe** [comment]
Nice catch, I'll fix it in a follow up, sorry for missing that!

**Pull request merged**
